torchinductor/ir.py

```diff
@@ -582,6 +657,18 @@ def fn(index):
 
             return Pointwise.create(device, dst_dtype, fn, ranges)
 
+        if (
+            V.graph.sizevars.size_hint(reduction_numel)
```

Comment:
I wonder if this is a good idea. If the reduction dim suddenly changes with dynamic shapes, it'll probably be quite bad if we unroll a massive reduction.

Perhaps we should only do this if the reduction dim is static?

Reply:
This won't happen due to the guard.  

"Is adding guards ok for optimizations?" is a reasonable question though.  We do it in some other places too.